src/components/ErrorBoundary/withErrorBoundary.js

```diff
@@ -1,17 +1,17 @@
 /**
  * @flow
- * @file withErrorBoundary HOC
+ * @file withErrorBoundary HOC which adds error boundaries as well as error error logging
  * @author Box
  */
 
 import * as React from 'react';
 import ErrorBoundary from './ErrorBoundary';
 
-const withErrorBoundary = (WrappedComponent: React.ComponentType<any>) =>
+const withErrorBoundary = (errorOrigin: ErrorOrigins) => (WrappedComponent: React.ComponentType<any>) =>
     // $FlowFixMe doesn't know about forwardRef (https://github.com/facebook/flow/issues/6103)
     React.forwardRef((props: Object, ref: React.Ref<any>) => (
-        <ErrorBoundary>
-            <WrappedComponent {...props} ref={ref} />
+        <ErrorBoundary {...props} ref={ref} errorOrigin={errorOrigin}>
```

Comment:
Moving the `ref` from the wrapped component to the error boundary is likely going to cause bugs.

Reply:
The ref is being forwarded there as well

Reply:
Ah I see what you mean, moving it back to the wrappedComponent